### PR TITLE
Persist dataset access properties when using H5Fstart_swmr_write - merge to 1.12 branch

### PR DESCRIPTION
**Pull request by fortnern**
Persist DAPL properties during H5Fstart_swmr_write()

### CODE REVIEW

**Comment by fortnern** [comment]
All tests passed except for codespell (spelling errors in files that this PR doesn't change): https://github.com/HDFGroup/hdf5/actions/runs/2878540588

**Review by derobins — APPROVED** [review]

File: src/H5Fint.c [hunk 1]
```diff
@@ -3800,6 +3854,14 @@ H5F__start_swmr_write(H5F_t *f)
     if (obj_paths)
         H5MM_xfree(obj_paths);
 
+    /* Free access property lists */
+    if (obj_apl_ids) {
+        for (u = 0; u < grp_dset_count; u++)
+            if (obj_apl_ids[u] != H5P_DEFAULT && obj_apl_ids[u] >= 0 && H5I_dec_ref(obj_apl_ids[u]) < 0)
+                HGOTO_ERROR(H5E_ATOM, H5E_CANTDEC, FAIL, "decrementing property list ID failed")
```
Comment (jhendersonHDF) [inline src/H5Fint.c]: Should be a DONE_ERROR instead of a GOTO since this is past the done label

Comment (derobins) [reply src/H5Fint.c]: This is also a problem in develop and needs to be fixed there, too

**Review by byrnHDF — APPROVED** [review]

File: src/H5Fint.c (hunk 1 above)
Comment (fortnern) [reply src/H5Fint.c]: Done and (h)done

**Review by derobins — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**